src/Symfony/Component/Form/Tests/Extension/Core/Type/ChoiceTypeTest.php

```diff
@@ -1505,4 +1476,65 @@ public function testCustomChoiceTypeDoesNotInheritChoiceLabels()
         // In this case the 'choice_label' closure returns null and not the closure from the first choice type.
         $this->assertNull($form->get('subChoice')->getConfig()->getOption('choice_label'));
     }
+
+    /**
+     * @dataProvider simpleWidgetsProvider
+     */
+    public function testSubmitChoicesWithSimpleWidgets($widget)
+    {
+        $form = $this->factory->create('Symfony\Component\Form\Extension\Core\Type\ChoiceType', null, array(
+            'widget' => $widget,
+            'multiple' => false,
+            'choices' => $this->choices,
+        ));
+
+        $form->submit('b');
+
+        $this->assertEquals('b', $form->getData());
+        $this->assertEquals('b', $form->getViewData());
+    }
+
+    /**
+     * @dataProvider simpleWidgetsProvider
+     */
+    public function testSubmitMultipleChoicesWithSimpleWidgets($widget)
+    {
+        $form = $this->factory->create('Symfony\Component\Form\Extension\Core\Type\ChoiceType', null, array(
+            'widget' => $widget,
+            'multiple' => true,
+            'choices' => $this->choices,
+        ));
+
+        $form->submit('a,b');
+
+        $this->assertEquals(array('a', 'b'), $form->getData());
+        $this->assertEquals('a,b', $form->getViewData());
+    }
+
+    /**
+     * @dataProvider simpleWidgetsProvider
+     */
+    public function testSubmitMultipleChoicesDelimiterAndTrimWithSimpleWidgets($widget)
+    {
+        $form = $this->factory->create('Symfony\Component\Form\Extension\Core\Type\ChoiceType', null, array(
+            'widget' => $widget,
+            'multiple' => true,
+            'delimiter' => '|',
+            'trim' => true,
+            'choices' => $this->choices,
+        ));
+
+        $form->submit('a| b ');
+
+        $this->assertEquals(array('a', 'b'), $form->getData());
+        $this->assertEquals('a|b', $form->getViewData());
+    }
+
+    public function simpleWidgetsProvider()
+    {
+        return array(
+            array('text'),
+            array('hidden'),
+        );
+    }
```

Comment:
There should also be tests checking `ChoiceType` options behaviour with widget `text` and `hidden`.

Including `choice_label`, `choice_value`, `placeholder`, `group_by`, and `preferred_choices`.